src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * This class encapsulated the management of local storage for Duke.
+ */
+public class Storage {
+
+    private final String SAVE_FILE_LOCATION;
+
+    /**
+     * Constructor for Storage object, requires the path to
+     * where the save file should be located.
+     *
+     * @param saveFileLocation
+     */
+    public Storage(String saveFileLocation) {
+        this.SAVE_FILE_LOCATION = saveFileLocation;
+    }
+
+    /**
+     * This method updates the local save file with the latest
+     * list of tasks. It is called whenever a task is added/deleted/completed.
+     *
+     * @param tasks The latest list of tasks.
+     */
+    public void updateSave(List<Task> tasks) {
+        String data = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            data += tasks.get(i).toStringSave() + "\n";
+        }
+        try {
+            FileOutputStream fos = new FileOutputStream(SAVE_FILE_LOCATION, false);
+            byte[] b = data.getBytes();
+            fos.write(b);
+            fos.close();
+        } catch (IOException ex) {
+            System.out.println("Error: Unable to save latest task. Please try again \n" + ex.getMessage());
+        }
+    }
+
+    /**
+     * This method is run once whenever the program is started up.
+     * It loads the previous save or creates a new save file if it is
+     * the first time the program is run.
+     *
+     * @return The array list from the previous save.
+     */
+    public ArrayList<Task> loadSave() {
+        try {
+            File saveFile = new File(SAVE_FILE_LOCATION);
+            Scanner sc = new Scanner(saveFile);
+            ArrayList<Task> tasks = new ArrayList<>();
+            while (sc.hasNextLine()) {
+                Task newTask;
+                String[] taskDetails = sc.nextLine().split(">");
+                String name = taskDetails[2];
+                boolean done = taskDetails[1].equals("1");
+                switch (taskDetails[0]) {
+                    case "T":
```

Comment:
The `case` in `switch` statements should be at the same level of indentation as the `switch` itself. You can refer to the module dashboard or online on how to change this in IntelliJ.